alibuild_helpers/build.py

```diff
@@ -405,10 +405,11 @@ def doBuild(args, parser):
 
   # Clone/update repos
   for p in [p for p in buildOrder if "source" in specs[p]]:
-    updateReferenceRepoSpec(args.referenceSources, p, specs[p], args.fetchRepos)
+    if not args.remoteStore or args.forceCloneRepos:
+      updateReferenceRepoSpec(args.referenceSources, p, specs[p], args.fetchRepos)
 
     # Retrieve git heads
-    cmd = "git ls-remote --heads %s" % (specs[p]["reference"] if "reference" in specs[p] else specs[p]["source"])
+    cmd = "git ls-remote --heads %s" % (specs[p].get("reference", specs[p]["source"]))
```

Comment:
This line we can merge immediately if you have it in a separate PR.

Reply:
https://github.com/alisw/alibuild/pull/594